### DISCUSSION TITLE
Request for new features

### DISCUSSION THREAD

**Discussion by overthetop78** [86Box/86Box#3058] [Ideas]
Hello,
In 1992 I had an interTAN brand pc (Tandy in France and probably Radioshack elsewhere) containing an AT motherboard in a desktop box with an Intel 80286 processor at 16Mhz (8Mhz without turbo), a PHOENIX BIOS. 1024 kB of RAM, a 40 MB IDE hard drive and a 3 "1/2 1.44 MB floppy drive. The integrated video chipset was a 256-color VGA. I can't find any way to redo this PC since the various upgrades day integrating chipsets NEAT, SCAT, MCA .... The PC I had was a simple AT (with extension ISA 16 and ISA 8). It had MS-DOS 5.0 and Tandy Deskmate 3. I can not find any photo of the PC that I had, unfortunately. I still have the MS-DOS "Quick Reference" Guide with the InterTAN mark (Tandy / RadioShack) but no other info. I possibly still have the PC manual but it is far away, at home my mom and I don't know if she didn't throw it all away.

I recovered in 1994 the PC of my father which was a computer assembled by a professional (without brand) it had an AT / ATX motherboard in a "Medium Tower" case, an AMI BIOS, an AMD am386 SX 25 Mhz processor (8 Mhz without turbo), 4096 KB of RAM. ISA 16 and ISA 8 expansion ports. An 80 MB IDE hard drive, a 3 "1/2 1.44 MB floppy drive. The integrated video chipset was a Trident TVGA. But I cannot find the possibility of doing the same under 86Box He had MS-DOS 5.0, then 6.0 then 6.2 and Windows 3.1.

1) I would like you to be able to integrate its configurations in 86Box because currently, the ISA 286 Phoenix does not support more than 8Mhz and there is no ISA 386 AMI supporting a 386 SX 25 Mhz.

2) There are still problems with the arrow keys making numbers instead of remaining as arrow keys. At the base, the arrow keys are quite equivalent to the keys of the numeric keypad but only the numeric keypad allows to change into numeric keys when the numeric lock is activated. So the arrow keys and the numeric keypad should be independent, in my opinion.

3) The PCs of the time had a turbo touch which allows to boost the speed of the processor. Would it be possible to integrate in 86Box the choice of the real speed and the turbo speed and to have a combination of keys to switch between Normal and Turbo (or in the menu of 86Box). Otherwise, in the same principle, integrate each CPU with references 386 SX 8/25, 8/16, 8/33. It's not important, it's just to get closer to what was happening.

4) Adding device:
There is the CD player, will we see the CD burner and DVD player?
Would it be possible to have the choice of the brand name of the CD device, and hard drives? Initially purely aesthetic but that too, via proprietary ROMs, to run certain applications of the associated brand. (Seagate, Maxtor, Quantum, NEC etc ..) when these are waiting for their specific peripherals.
- JAZ, LS-120, Spark players?
- No more video card and chipset?
- If someone has the courage to get started with video capture cards (TV card, FM card, capture card ... which could be connected to an audio-video player? Has audio-video streams or applications audio-video stream.)
- Different printers?
- Added scanner?
- Adding a network card?
- Addition of Modem card and chipset?
- Addition of a specific controller / Joystick (I have a Boeder controller in DB15 Game Port having an analog joystick of the directional cross type, 4 triggers (simple buttons), 6 action buttons (A, B, C, X, Y , Z) and 4 Buttons in the middle (like Start, Select).)
The choice of the Sidewinder Gamepad does not want to work for me, it is recognized in Windows 9x but no associated button on my Xbox controller reacts in the controller control panel or in the Sidewinder tool. This is well configured in 86Box and everything works correctly with another choice of controller.
- Mouse with more than 3 buttons + wheel? Trackball?
- Microphone?
- Webcam?

5) A question that I also ask myself, the possibility of choosing the RAM sticks according to the locations available on the associated motherboard. Useful? I do not know. Compared to the motherboard ROM, can this affect the speed of this memory compared to the ROM? I do not know.

Thanks for reading me. And I also know that you have to make different messages for each request, but I preferred to put everything together, hoping to be better understood and that you can take what you want into consideration.
For the arrow keys problem. There is already a message, I thought the problem was resolved but it still is not.

Ah yes! I saw that there was no longer a choice of 86Box version depending on our CPU. Is everything gathered on the same version?
Is it for keyboard repairs maybe in the Experimental version, rather than the Stable or Debug version?

**Comment by bachimusprime** [comment]
Just want to note that we already have printers and network cards implemented.

**Comment by OBattler** [comment]
> I would like you to be able to integrate its configurations in 86Box because currently, the ISA 286 Phoenix does not support more than 8Mhz and there is no ISA 386 AMI supporting a 386 SX 25 Mhz.

What exactly do you mean by this?

> There are still problems with the arrow keys making numbers instead of remaining as arrow keys. At the base, the arrow keys are quite equivalent to the keys of the numeric keypad but only the numeric keypad allows to change into numeric keys when the numeric lock is activated. So the arrow keys and the numeric keypad should be independent, in my opinion.

On AT+, they already are independent, on XT, they can't be since an XT keyboard only has the numeric keypad.

> The PCs of the time had a turbo touch which allows to boost the speed of the processor. Would it be possible to integrate in 86Box the choice of the real speed and the turbo speed and to have a combination of keys to switch between Normal and Turbo (or in the menu of 86Box). Otherwise, in the same principle, integrate each CPU with references 386 SX 8/25, 8/16, 8/33. It's not important, it's just to get closer to what was happening.

I'm still thinking of how to properly implement this without causing a timer mess-up.

> Adding device:
There is the CD player, will we see the CD burner and DVD player?
Would it be possible to have the choice of the brand name of the CD device, and hard drives? Initially purely aesthetic but that too, via proprietary ROMs, to run certain applications of the associated brand. (Seagate, Maxtor, Quantum, NEC etc ..) when these are waiting for their specific peripherals.

The currently emulated CD-ROM drives can also read DVD's. Brand-specific commands are not feasible due to no documentation available.

> JAZ, LS-120, Spark players?

If I get the stuff I need (SCSI INQUIRY, MODE SENSE, and ATAPI IDENTIFY dumps), then it's doable.

> No more video card and chipset?

And yet, we have people requesting stuff like 8514/a and so on, and EISA is coming in v4.0. I'm not exactly sure why we should put a permanent stop adding video cards and chipsets.

> If someone has the courage to get started with video capture cards (TV card, FM card, capture card ... which could be connected to an audio-video player? Has audio-video streams or applications audio-video stream.)

I've had the same idea as well, but it would require a lot of work, especially if one were to emulate TV signals accurately.

> Different printers?

I'd love to add HP PCL printer emulate using GhostPCL, but I can't figure out how to use that.

> Added scanner?

I might do that some day.

> Adding a network card?

Which exactly? There's a lot of them that we don't yet emulate.

> Addition of Modem card and chipset?

Modem emulation is in my plans for v4.0.

> Addition of a specific controller / Joystick (I have a Boeder controller in DB15 Game Port having an analog joystick of the directional cross type, 4 triggers (simple buttons), 6 action buttons (A, B, C, X, Y , Z) and 4 Buttons in the middle (like Start, Select).)
> The choice of the Sidewinder Gamepad does not want to work for me, it is recognized in Windows 9x but no associated button on my Xbox controller reacts in the controller control panel or in the Sidewinder tool. This is well configured in 86Box and everything works correctly with another choice of controller.

We'd need documentation of those joystick types.

> Mouse with more than 3 buttons + wheel? Trackball?

Again the problem is finding the documentation of such mice. Also, how exactly does a trackball mouse differ from a normal one to a driver? There should be no difference.

> Microphone?

There is code for that in the emulated Sound Blaster 16, but currently there's no framework for audio input.

> Webcam?

Maybe at some point in the not so near future, when we're more knowledgeable in USB devices.

> A question that I also ask myself, the possibility of choosing the RAM sticks according to the locations available on the associated motherboard. Useful? I do not know. Compared to the motherboard ROM, can this affect the speed of this memory compared to the ROM? I do not know.

I could add RAM stick selection, but it would just overcomplicate everything. There's already enough users complaining that the emulator is too hard to use, I don't want to make it even harder.

> Ah yes! I saw that there was no longer a choice of 86Box version depending on our CPU. Is everything gathered on the same version?

Optimized versions are no longer provided because they are buggy.

> Is it for keyboard repairs maybe in the Experimental version, rather than the Stable or Debug version?

If you mean the keyboard fixes, as in, the big keyboard controller etc. rewrite, that's currently in my feature/machine_and_kb branch but is very unfinished.

**Reply by robman501a** [reply]
@OBattler  For modem emulation, I suppose you can make the emulated modem make the dialing tones, handshake sounds, and once it says "connected" you could have the guest OS connect to the network on the host OS.

**Comment by overthetop78** [comment]
I would like you to be able to integrate its configurations in 86Box because currently, the ISA 286 Phoenix does not support more than 8Mhz and there is no ISA 386 AMI supporting a 386 SX 25 Mhz.

What exactly do you mean by this?

I meant that 80286 AT type CPUs cannot go higher than 8Mhz. However, I had a Tandy AT 80286 at 16Mhz (8Mhz Normal, 16Mhz Turbo)

Ditto for the 80386 SX 25 Mhz, there is none with an AMI brand BIOS. My father had this model (an AMD am386 SX 25 Mhz in Turbo mode [8 Mhz in normal mode])

Would it be possible to add or modify the 286 ATs so that they can run at 16 Mhz and the 386 SX with the choice of AMD CPU at 25Mhz and even 33Mhz and 40Mhz on motherboards with AMI BIOS?
I am sorry to respond so late.
Thank you for all your answers. I see that you have a lot of desire, a lot of project but unfortunately little documentation.
Perhaps it is possible to ask certain manufacturers of its old equipment if they agree to provide documentation.
